client/components/domains/register-domain-step/index.jsx

```diff
@@ -31,19 +32,53 @@ var wpcom = require( 'lib/wp' ).undocumented(),
 	abtest = require( 'lib/abtest' ).abtest;
 
 // max amount of domain suggestions we should fetch/display
-var SUGGESTION_QUANTITY = 10,
+const SUGGESTION_QUANTITY = 10,
 	INITIAL_SUGGESTION_QUANTITY = 2;
 
+const analytics = analyticsMixin( 'registerDomain' );
+
+let searchQueue = [],
+	searchStackTimer = null;
+
+function processSearchQueue() {
```

Comment:
What do you think about putting these three new functions (`processSearchQueue`, `reportSearch` and `enqueueSearch`) and the variables associated with them (`searchQueue` and `searchStackTimer`) to a separate file (can be in the same directory). You wouldn't even have to expose the `reportSearch` function :) That seems like a less cluttered approach - this file is long and complicated already as it is :grinning: And no one would be tempted to access those variables directly :P